src/configuration.ts

```diff
@@ -13,164 +13,159 @@ import { PathResolver } from "./resolvers/path-resolver";
+        settings,
+        workspace.workspaceFolders[index].name
+      );
+
+      resourcesSettings.splice(index, 0, settings);
     }
 
-    /**
-     * Attempt to find the root path for a workspace or resource
-     * @param resource
-     */
-    private resolveRootPath(resource: Uri) {
-        // try to get a valid folder from resource
-        let folder = workspace.getWorkspaceFolder(resource);
-
-        // one last safety check
-        return folder ? folder.uri.fsPath : "";
+    // update settings from config
+    let settings: Settings = {
+      resources: resourcesSettings,
+      snifferMode: this.config.get("snifferMode", "onSave"),
+      snifferShowSources: this.config.get("snifferShowSources", false),
+      snifferTypeDelay: this.config.get("snifferTypeDelay", 250),
+      debug: this.debug,
+    };
+    this.logger.logInfo("CONFIGURATION", settings);
+    return settings;
+  }
+
+  /**
+   * Attempt to find the root path for a workspace or resource
+   * @param resource
+   */
+  private resolveRootPath(resource: Uri) {
+    // try to get a valid folder from resource
+    let folder = workspace.getWorkspaceFolder(resource);
+
+    // one last safety check
+    return folder ? folder.uri.fsPath : "";
+  }
+
+  /**
+   * Get correct executable path from resolver
+   * @param settings
+   */
+  protected async resolveCBFExecutablePath(
+    settings: ResourceSettings
+  ): Promise<ResourceSettings> {
+    if (settings.executablePathCBF === null || settings.executablePathCBF === "") {
+      let executablePathResolver = new PathResolver(settings, "phpcbf");
+      settings.executablePathCBF = await executablePathResolver.resolve();
+    } else if (
+      !path.isAbsolute(settings.executablePathCBF) &&
+      settings.workspaceRoot !== null
+    ) {
+      settings.executablePathCBF = path.join(
+        settings.workspaceRoot,
+        settings.executablePathCBF
+      );
     }
-
-    /**
-     * Get correct executable path from resolver
-     * @param settings
-     */
-    protected async resolveCBFExecutablePath(
-        settings: ResourceSettings
-    ): Promise<ResourceSettings> {
-        if (settings.executablePathCBF === null) {
-            let executablePathResolver = new PathResolver(settings, "phpcbf");
-            settings.executablePathCBF = await executablePathResolver.resolve();
-        } else if (
-            !path.isAbsolute(settings.executablePathCBF) &&
-            settings.workspaceRoot !== null
-        ) {
-            settings.executablePathCBF = path.join(
-                settings.workspaceRoot,
-                settings.executablePathCBF
-            );
-        }
-        return settings;
+    return settings;
+  }
+
+  /**
+   * Get correct executable path from resolver
+   * @param settings
+   */
+  protected async resolveCSExecutablePath(
+    settings: ResourceSettings
+  ): Promise<ResourceSettings> {
+    if (settings.executablePathCS === null || settings.executablePathCS === "") {
```

Comment:
I think this can probably just be if (!settings.executablePathCS)

Reply:
That's a good point, I will check.